### PR TITLE
feat: add reverse_a_queue.cpp (issue #1851)

### PR DESCRIPTION
**Pull request by sng125**
#### Description of Change

Added a queue reversing algorithm that satisfies issue [#1851](https://github.com/TheAlgorithms/C-Plus-Plus/issues/1851)

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1891"></a>

### CODE REVIEW

File: data_structures/reverse_a_queue.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * follows Last In First Out (LIFO). By popping every element of a queue into a
+ * stack, the last element inserted will be the first element to leave the
+ * stack, and the first element inserted will be the last element to leave the
+ * stack.
+ */
+
+#include <cassert>   // for asserting during testing
+#include <iostream>  // for IO operations
+#include <queue>     // for testing stl queue
+#include <stack>     // for storing reversed queue
```
Comment (Panquesito7) [inline data_structures/reverse_a_queue.cpp:17]:
```suggestion
#include <cassert>   /// for assert
#include <iostream>  /// for IO operations
#include <queue>     /// for testing stl queue
#include <stack>     /// for storing reversed queue
```

File: data_structures/reverse_a_queue.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+/**
+ * @brief Main method
+ * @returns returns 0 on exit
+ */
+int main() {
+    test_reverse_queue();
+    return 0;
+}
```
Comment (Panquesito7) [inline data_structures/reverse_a_queue.cpp:70]: Missing an endline. Without this endline, some IDEs and configurations can have issues.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: data_structures/reverse_a_queue.cpp (hunk 2 above)
Comment (sng125) [reply data_structures/reverse_a_queue.cpp]: I have fixed this problem.

File: data_structures/reverse_a_queue.cpp (hunk 1 above)
Comment (sng125) [reply data_structures/reverse_a_queue.cpp]: I have fixed this issue.

**Pull request closed without merging**